test/lit/passes/extract-function.wast

```diff
@@ -0,0 +1,72 @@
+;; RUN: foreach %s %t wasm-opt --extract-function=foo -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --extract-function --pass-arg=extract-function@foo -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --extract-function-index=0 -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --extract-function-index --pass-arg=extract-function-index@0 -S -o - | filecheck %s
+
+;; CHECK:      (module
+;; CHECK-NEXT:  (type $none_=>_none (func))
+;; CHECK-NEXT:  (import "env" "bar" (func $bar))
+;; CHECK-NEXT:  (export "foo" (func $foo))
+;; CHECK-NEXT:  (func $foo
+;; CHECK-NEXT:   (call $bar)
+;; CHECK-NEXT:  )
+;; CHECK-NEXT: )
```

Comment:
These are manually-created, correct? I worry about us accumulating a lot of those and having a bad time some day when we need to update them. Adding `foreach` seems like it would increase the amount of such test code... How hard would it be to support auto updating in these eventually?

Reply:
Yes, these are manually created for now. I think it would be reasonable to update the auto update script in the near future to handle multiple multiple modules in a file and also to give it an option to emit checks on a per-module rather than a per-function basis.